### PR TITLE
[New Rules] DPKG Execution/Installation

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This PR adds two new rules related to the installation of a DPKG package and the execution of the DPKG utility via an unusual parent.

### Unusual DPKG Execution
This rule detects the execution of the DPKG command by processes not associated with the DPKG package manager. The DPKG command is used to install, remove, and manage Debian packages on a Linux system. Attackers can abuse the DPKG command to install malicious packages on a system.

Only TPs in my testing stack, 0 hits in telemetry last 90d.

image

### DPKG Package Installed by Unusual Parent Process
This rule detects the installation of a Debian package (dpkg) by an unusual parent process. The dpkg command is used to install, remove, and manage Debian packages on a Linux system. Attackers can abuse the dpkg command to install malicious packages on a system.

It uses the new_terms rule type to minimize FPs across hosts.

100+ hits in telemetry last 90d, but by using the new_terms rule type this should probably decrease. Also, this rule is meant to be relatively noisy, as it is a low confidence/risk rule. It grants an extra layer of visibility into .deb package persistence.

image

### CODE REVIEW

File: rules/linux/persistence_dpkg_package_installation_from_unusual_parent.toml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        "OS: Linux",
+        "Use Case: Threat Detection",
+        "Tactic: Persistence",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "new_terms"
+query = '''
+host.os.type:linux and event.category:process and event.type:start and event.action:exec and process.name:dpkg and
+process.args:"-i"
```
Comment (Aegrah) [inline rules/linux/persistence_dpkg_package_installation_from_unusual_parent.toml:61]:
```suggestion
process.args:("-i" or "--install")
```

**Review by terrancedejesus — APPROVED** [review]
Very nice additions ✅

File: rules/linux/persistence_dpkg_unusual_execution.toml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        "Use Case: Threat Detection",
+        "Tactic: Persistence",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.type == "start" and event.action == "exec" and
+process.executable : "/var/lib/dpkg/info/*" and process.session_leader.name != null and
+process.group_leader.name != null and not (
```
Comment (terrancedejesus) [inline rules/linux/persistence_dpkg_unusual_execution.toml:60]: Wonder if the unit test failure is here?

```
and not (
  process.parent.name in ("dpkg", "dpkg-reconfigure") or
  process.session_leader.name == "dpkg" or
  process.group_leader.name: ("dpkg" or null)
)
```

Comment (Aegrah) [reply rules/linux/persistence_dpkg_unusual_execution.toml:60]: I fixed it, it was an illogical MITRE sequence.

**Review by imays11 — APPROVED** [review]

**Pull request merged**